### PR TITLE
[feature] fia support sliding windows

### PR DESCRIPTION
**Pull request by NSDie**
fia support sliding windows

### What this PR does / why we need it?
Enable fia to support sliding window function and adapt to the Gemma3 model.

### Does this PR introduce _any_ user-facing change?

modify the `forward_fused_infer_attention` function in the `attention_v1.py`.

### How was this patch tested?

- vLLM version: release/v0.13.0
- vLLM main: https://github.com/vllm-project/vllm/commit/ad32e3e19ccf0526cb6744a5fed09a138a5fb2f9

### CODE REVIEW

File: vllm_ascend/attention/attention_v1.py [hunk 1]
```diff
@@ -363,6 +363,12 @@ def __init__(
         self.num_queries_per_kv = self.num_heads // self.num_kv_heads
         self.key_cache = None
         self.value_cache = None
+        self.swa_mask = None
+        if self.sliding_window is not None:
+            mask = torch.ones(2048, 2048, dtype=torch.bool)
+            triu_mask = torch.triu(mask, diagonal=1).to("npu")
+            tril_mask = torch.tril(mask, -self.sliding_window).to("npu")
+            self.swa_mask = triu_mask + tril_mask
```
Comment (weijinqian0) [inline vllm_ascend/attention/attention_v1.py]: may need move this to AttnMaskBuilder for cache and use only one same mask.
